src/preprocess_info.cc

```diff
@@ -158,7 +195,236 @@ Napi::Value PreProcessInfo::GetMeanVariant(const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
+
+  if (info.Length() != 2) {
+    Napi::TypeError::New(env, "Wrong number of arguments")
+        .ThrowAsJavaScriptException();
+    return;
+  }
+
+  if (!info[0].IsNumber()) {
+    Napi::TypeError::New(env, "Wrong type of arguments")
+        .ThrowAsJavaScriptException();
+    return;
+  }
+
+  if (!info[1].IsObject()) {
+    Napi::TypeError::New(env, "Wrong type of arguments")
+        .ThrowAsJavaScriptException();
+    return;
+  }
+
+  size_t index = info[0].ToNumber().Int32Value();
+  Napi::Object newChannel = info[1].ToObject();
+
+  ie::PreProcessInfo& preInfo = _input_info->getPreProcess();
+
+  size_t numberOfChannels = preInfo.getNumberOfChannels();
+
+  if (numberOfChannels == 0) {
+    Napi::Error::New(env, "accessing pre-process when nothing was set.")
+        .ThrowAsJavaScriptException();
+    return;
+  };
+
+  if (index >= numberOfChannels || index < 0) {
+    std::string errorLog =
+        "pre process index " + std::to_string(index) + " is out of bounds.";
+    Napi::Error::New(env, errorLog).ThrowAsJavaScriptException();
+    return;
+  };
+
+  ie::PreProcessChannel::Ptr channel = preInfo[index];
+  if (newChannel.Has("stdScale")) {
+    if (newChannel.Get("stdScale").IsNumber()) {
+      channel->stdScale = newChannel.Get("stdScale").ToNumber().FloatValue();
+    } else {
+      Napi::TypeError::New(env, "Wrong stdScale").ThrowAsJavaScriptException();
+    }
+  }
+
+  if (newChannel.Has("meanValue")) {
+    if (newChannel.Get("meanValue").IsNumber()) {
+      channel->meanValue = newChannel.Get("meanValue").ToNumber().FloatValue();
+    } else {
+      Napi::TypeError::New(env, "Wrong meanValue").ThrowAsJavaScriptException();
+    }
+  }
+
+  if (newChannel.Has("meanData")) {
+    if (!newChannel.Get("meanData").IsObject()) {
+      Napi::TypeError::New(env, "Wrong meanData").ThrowAsJavaScriptException();
+      return;
+    }
+
+    Napi::Object meanData = newChannel.Get("meanData").ToObject();
+
+    if (!meanData.Has("desc") || !meanData.Has("data")) {
+      Napi::TypeError::New(env, "Wrong meanData").ThrowAsJavaScriptException();
+      return;
+    }
+
+    if (!meanData.Get("desc").IsObject()) {
+      Napi::TypeError::New(env, "Wrong desc").ThrowAsJavaScriptException();
+      return;
+    }
+
+    if (!meanData.Get("data").IsArrayBuffer()) {
+      Napi::TypeError::New(env, "Wrong data").ThrowAsJavaScriptException();
+      return;
+    }
+
+    Napi::Object desc = meanData.Get("desc").ToObject();
+    if (!checkDesc(desc)) {
+      Napi::TypeError::New(env, "Wrong desc").ThrowAsJavaScriptException();
+      return;
+    }
+
+    Napi::ArrayBuffer data =
+        Napi::ArrayBuffer::ArrayBuffer(env, meanData.Get("data"));
+    std::string precision_string = desc.Get("precision").ToString();
+    std::string layout_string = desc.Get("layout").ToString();
+    Napi::ArrayBuffer dims =
+        Napi::ArrayBuffer::ArrayBuffer(env, desc.Get("dims"));
```

Comment:
You should validate whether `desc` has `dims`.

Reply:
I would suggest the `dims` to be an `Array` instead of `ArrayBuffer`.

Reply:
In my investigation, I find it is hard to use the normal Array in C++. The data type Array in N-API even does not have API to get elements. Meanwhile, I find that the official says that Napi::TypedArray and Napi::ArrayBuffer can be used for transferring large amounts of data from JavaScript to the native side. Thus, I choose the ArrayBuffer to finish the implementation.

Reply:
Please refer to [Napi::Array](https://github.com/nodejs/node-addon-api/blob/master/doc/basic_types.md#array). You can get its length by [Napi::Array::Length()](https://github.com/nodejs/node-addon-api/blob/master/doc/basic_types.md#length)
and access the elements by indexing `array[i]` refer to [Napi::Object::operator[] (uint32_t index)](https://github.com/nodejs/node-addon-api/blob/master/doc/object.md#operator)

Reply:
Thanks for your suggestion. I will complete it soon.

Reply:
> You should validate whether `desc` has `dims`.

It has been completed by the function `checkDesc()` , too.

Reply:
Done.